src/EditorFeatures/Core/Implementation/IntelliSense/Helpers.cs

```diff
@@ -72,7 +140,60 @@ internal static IEnumerable<object> BuildClassifiedTextElements(ImmutableArray<T
             return paragraphs;
         }
 
-        internal static object CreateParagraphFromLines(IReadOnlyList<ClassifiedTextElement> lines)
+        private static void NavigateToQuickInfoTarget(string navigationTarget, Document document, IEnumerable<Lazy<IStreamingFindUsagesPresenter>> streamingPresenters)
+        {
+            var navigateToLinkService = document.Project.Solution.Workspace.Services.GetRequiredService<INavigateToLinkService>();
+            if (Uri.TryCreate(navigationTarget, UriKind.Absolute, out var absoluteUri))
+            {
+                navigateToLinkService.TryNavigateToLinkAsync(absoluteUri, CancellationToken.None);
+                return;
+            }
+
+            SymbolKeyResolution resolvedSymbolKey;
+            try
+            {
+                resolvedSymbolKey = SymbolKey.Resolve(navigationTarget, document.Project.GetCompilationAsync(CancellationToken.None).WaitAndGetResult(CancellationToken.None), cancellationToken: CancellationToken.None);
+            }
+            catch
+            {
+                // Ignore symbol resolution failures. It likely is just a badly formed URI.
+                return;
```

Comment:
def not liking this. 
1. symbolkey shoudn't be throwing.
2. not happy about all this sync blocking.  can we not be async here?

Worst case, when someone navigates, we can do a TWD and make it cancellable if something goes off the rails.

Reply:
> not happy about all this sync blocking. can we not be async here?

Limited by the editor API. @cgunderman how would you handle this?

Reply:
note: sync blocking is ok with me.  it's more the lack of cancellability.  That should hopefully be something we can fix up with a TWD.

Reply:
I think the only option is to do whatever the equivalent of a `Dispatcher.InvokeAsync()` is in this layer. From the IDE side, we usually use JTF.RunAsync(). I understand this is the portable editor features, so, there likely is no dispatcher.

For my own edification, would it be preferable to have actions like this be async? I didn't think it was necessary originally because we're de facto going to have to InvokeAsync() somewhere but I acknowledge things are a bit more constrained for Roslyn since this layer has to work on Mac.

Reply:
I'd be fine with this being async (which honestly makes sense to me given that any of thse clicks might cause major work to happen).  Or, if sync, at least have it be under a TWD.